src/lc3_vm/virtual_machine.rs

```diff
@@ -1,10 +1,23 @@
-use thiserror::Error;
-
 use super::{
     flags::ConditionFlags,
     opcodes::{Opcode, OpcodeError},
+    traps::Trap,
+};
+use nix::sys::{
+    select,
+    time::{TimeVal, TimeValLike},
+};
+use std::{
+    fmt::Debug,
+    fs::File,
+    io::{self, Read, Write},
+    os::fd::AsFd,
 };
+use thiserror::Error;
+
 const MEMORY_MAX: usize = 1 << 16;
+const MR_KBSR: u16 = 0xFE00;
+const MR_KBDR: u16 = 0xFE02;
```

Comment:
I found this constant a bit hard to read. Maybe some documentation links explaining it could be placed as a comment on top of it.

Very very minor however.

Reply:
You are right I'm planning to add all references to the README file

Reply:
It's defined exactly like that in the documentation for the LC3 so I don't think it's necessary, though it can use a comment next to it.